pkg/controller/job/job_controller.go

```diff
@@ -504,18 +509,7 @@ func (jm *Controller) deleteJob(logger klog.Logger, obj interface{}) {
 			return
 		}
 	}
-	// Listing pods shouldn't really fail, as we are just querying the informer cache.
-	selector, err := metav1.LabelSelectorAsSelector(jobObj.Spec.Selector)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("parsing deleted job selector: %v", err))
-		return
-	}
-	pods, _ := jm.podStore.Pods(jobObj.Namespace).List(selector)
-	for _, pod := range pods {
-		if metav1.IsControlledBy(pod, jobObj) && hasJobTrackingFinalizer(pod) {
-			jm.enqueueOrphanPod(pod)
-		}
-	}
+	jm.backupRemovePodFinalizers(jobObj)
```

Comment:
I'm not sure the name of the function reflects its role. Is it "backup" removal when called from `deleteJob`, as invoked currently on master branch?

Maybe we can consider naming it `cleanupPodFinalizers` (or `gcPodFinalizers`) + a comment next to the new invocation indicating that it is a backup invocation in `updateJob` (as done currently).